From 02d050e8d52da97e3701067f85d71e62622f7d88 Mon Sep 17 00:00:00 2001
From: Will Eccles <will@eccles.dev>
Date: Mon, 24 May 2021 12:48:51 -0400
Subject: [PATCH 2/2] cpsw: fix undefined function with PM disabled

When CONFIG_PM is not enabled, the function
davinci_mdio_update_dt_from_phymask is not defined. This patch fixes
this so that the build does not fail with CONFIG_PM disabled.

Signed-off-by: Will Eccles <will@eccles.dev>
---
 drivers/net/ethernet/ti/davinci_mdio.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/ti/davinci_mdio.c b/drivers/net/ethernet/ti/davinci_mdio.c
index 020ec0cc22c6..18334d6c77b1 100644
--- a/drivers/net/ethernet/ti/davinci_mdio.c
+++ b/drivers/net/ethernet/ti/davinci_mdio.c
@@ -480,6 +480,8 @@ static int davinci_mdio_runtime_resume(struct device *dev)
 	davinci_mdio_enable(data);
 	return 0;
 }
+#endif
+
 static void davinci_mdio_update_dt_from_phymask(u32 phy_mask)
 {
 	int i, len, skip;
@@ -567,7 +569,6 @@ static void davinci_mdio_update_dt_from_phymask(u32 phy_mask)
 		kfree(phy_id_property);
 	}
 }
-#endif
 
 #ifdef CONFIG_PM_SLEEP
 static int davinci_mdio_suspend(struct device *dev)
-- 
2.30.2

